### PR TITLE
add `DirectConnectionManager`

### PR DESCRIPTION
**Pull request by shouples**
## Summary of Changes

Closes https://github.com/confluentinc/vscode/issues/422.

This PR adds a basic connection manager class responsible for watching the experimental direct connection setting and setting the associated contextValue for the UI. This currently only controls the visibility of the "Add New Connection" action in the command palette, as well as the placeholder action in the Resources view in the nav area.

Toggle the direct connection experimental setting to see the action in the Resources view nav area:
image

> [!NOTE]  
> This branch is a "small" intermediate step to the next branch, which will be implementing **a lot** of the required functionality and refactoring for making the direct connections usable. This is the primarily to ensure the plumbing between the webview form and the sidecar is set up and some data storage prep work is ironed out.
> That said, **the only place you will see new direct connection information is in the logs / output channel until the next branch is merged**.

## Associated PRs
- https://github.com/confluentinc/vscode/pull/652
  - https://github.com/confluentinc/vscode/pull/636 👈 
    - https://github.com/confluentinc/vscode/pull/654

## Pull request checklist

Please check if your PR fulfills the following (if applicable):

##### Tests

- [x] Added new
- [x] Updated existing
- [ ] Deleted existing

##### Other

- [x] All new disposables (event listeners, views, channels, etc.) collected as  for eventual cleanup?

- [ ] Does anything in this PR need to be mentioned in the user-facing [CHANGELOG](https://github.com/confluentinc/vscode/blob/main/CHANGELOG.md) or [README](https://github.com/confluentinc/vscode/blob/main/public/README.md)?
- [x] Have you validated this change locally by [packaging](https://github.com/confluentinc/vscode/blob/main/README.md#packaging-steps) and installing the extension `.vsix` file?
  ```shell
  gulp clicktest
  ```

### CODE REVIEW

File: src/webview/direct-connect-form.html [hunk 1]
```diff
@@ -16,10 +16,11 @@ <h1>Set up a Direct Connection</h1>
       <template data-if="this.errorMessage()">
         <div class="info error" data-text="this.errorMessage()"></div>
       </template>
+
       <form data-on-submit="this.handleSubmit(event)">
         <div class="input-container">
           <div>
-            <label for="connection-label" class="label">Connection Label (optional)</label>
```
Comment (noeldevelops) [inline src/webview/direct-connect-form.html]: should this be required?

Comment (shouples) [reply src/webview/direct-connect-form.html]: Nope! Fixed in https://github.com/confluentinc/vscode/pull/636/commits/b33b3d2c1a14523e9ea6ad19da5fb6926c0f4f3c after some merge conflict cleanup

File: src/storage/resourceManager.ts
Comment (shouples) [inline src/storage/resourceManager.ts:1]: None of the changes in this file are actually used yet, just prep work for the following branch where we start storing `connectionId:ConnectionSpec`s as a Map and "rehydrate" the direct connections (by informing the sidecar of them) on extension activation.

File: src/directConnectManager.ts [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+      return { success: false, message: errorMessage };
+    }
+
+    // TODO(shoup): enable this in follow-on branch
+    // await getResourceManager().addDirectConnection(spec);
+
+    // TODO(shoup): refresh Resources view in follow-on branch
+
+    // `message` is hard-coded in the webview, so we don't actually use the connection object yet
+    return { success, message: JSON.stringify(connection) };
```
Comment (noeldevelops) [inline src/directConnectManager.ts:113]: 👍

**Review by noeldevelops — APPROVED** [review]
looks good!

**Pull request merged**